x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ApiKeyServiceTests.java

```diff
@@ -1250,6 +1386,108 @@ public void testGetCreatorRealm() {
+        authentication = Authentication.newRealmAuthentication(
+            AuthenticationTests.randomUser(),
+            AuthenticationTests.randomRealmRef(randomBoolean())
+        );
+        if (randomBoolean()) {
+            authentication.token();
+        }
+        assertThat(
+            Arrays.asList(ApiKeyService.getOwnersRealmNames(authentication.runAs(AuthenticationTests.randomUser(), realmRef))),
+            contains(realmRef.getName())
+        );
+        assertThat(
+            Arrays.asList(ApiKeyService.getOwnersRealmNames(authentication.runAs(AuthenticationTests.randomUser(), realmRef).token())),
+            contains(realmRef.getName())
+        );
+        // realm under domain
+        realmRef = AuthenticationTests.randomRealmRef(true);
+        authentication = Authentication.newRealmAuthentication(AuthenticationTests.randomUser(), realmRef);
+        assertThat(
+            Arrays.asList(ApiKeyService.getOwnersRealmNames(authentication)),
+            containsInAnyOrder(
+                realmRef.getDomain().realms().stream().map(realmIdentifier -> realmIdentifier.getName()).toArray(String[]::new)
+            )
+        );
+        assertThat(
+            Arrays.asList(ApiKeyService.getOwnersRealmNames(authentication.token())),
+            containsInAnyOrder(
+                realmRef.getDomain().realms().stream().map(realmIdentifier -> realmIdentifier.getName()).toArray(String[]::new)
+            )
+        );
+        // realm under domain run-as
+        authentication = Authentication.newRealmAuthentication(
+            AuthenticationTests.randomUser(),
+            AuthenticationTests.randomRealmRef(randomBoolean())
+        );
+        assertThat(
+            Arrays.asList(ApiKeyService.getOwnersRealmNames(authentication.runAs(AuthenticationTests.randomUser(), realmRef))),
+            containsInAnyOrder(
+                realmRef.getDomain().realms().stream().map(realmIdentifier -> realmIdentifier.getName()).toArray(String[]::new)
+            )
+        );
+        assertThat(
+            Arrays.asList(ApiKeyService.getOwnersRealmNames(authentication.runAs(AuthenticationTests.randomUser(), realmRef).token())),
+            containsInAnyOrder(
+                realmRef.getDomain().realms().stream().map(realmIdentifier -> realmIdentifier.getName()).toArray(String[]::new)
+            )
+        );
+        // API key authentication
+        final String apiKeyCreatorRealm = randomAlphaOfLengthBetween(2, 8);
+        authentication = Authentication.newApiKeyAuthentication(
+            AuthenticationResult.success(
+                AuthenticationTests.randomUser(),
+                Map.of(
+                    AuthenticationField.API_KEY_CREATOR_REALM_NAME,
+                    apiKeyCreatorRealm,
+                    AuthenticationField.API_KEY_ID_KEY,
+                    randomAlphaOfLength(20)
+                )
+            ),
+            randomAlphaOfLength(4)
+        );
```

Comment:
I suggest we use `AuthenticationTests#randomApiKeyAuthentication` for this setup.

Reply:
Yes, I made that method expose the creator realm name as an argument.